sdk/src/azure/core/az_span.c

```diff
@@ -554,25 +554,40 @@ void az_span_to_str(char* destination, int32_t destination_max_size, az_span sou
 
 /**
  * @brief Replace all contents from a starting position to an end position with the content of a
- * provided span
+ * provided span.
  *
- * @param destination src span where to replace content
- * @param start starting position where to replace
- * @param end end position where to replace
- * @param replacement content to use for replacement
+ * @remarks
+ *
+ * @param[in] destination src span where to replace content
+ * @param[in] start starting position where to replace
+ * @param[in] end end position where to replace
+ * @param[in] replacement content to use for replacement
+ * @param[in] url_encode when true, the replacement value will be url-encoded before using it to
+ * replace content
+ * @param[out] out_url_encode_size when url_encode is true, the size of the url-encoded replacement
+ * is recorded in this ptr. Using null when \p url_encode is fine, but it should not be null if \p
+ * url_encode is true.
  * @return AZ_NODISCARD az_span_replace
  */
 AZ_NODISCARD az_result _az_span_replace(
     az_span destination,
     int32_t current_size,
     int32_t start,
     int32_t end,
-    az_span replacement)
+    az_span replacement,
+    bool url_encode,
+    int32_t* out_url_encode_size)
 {
-  int32_t const replacement_size = az_span_size(replacement);
+  int32_t const replacement_size
+      = url_encode ? _az_span_url_encode_calc_length(replacement) : az_span_size(replacement);
   int32_t const replaced_size = end - start;
   int32_t const size_after_replace = current_size - replaced_size + replacement_size;
 
+  if (url_encode)
+  {
+    _az_PRECONDITION_NOT_NULL(out_url_encode_size);
+  }
```

Comment:
We don't want an extra check when preconditions are turned off. Change the condition so we don't need the `if`

```suggestion
  _az_PRECONDITION(!url_encode || (url_encode && out_url_encode_size != NULL));
```